src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureVirtualNetworkOperationSDK.java

```diff
@@ -0,0 +1,96 @@
+        this.scheduler = Schedulers.from(virtualNetworkExecutor);
+        this.regionName = regionName;
+    }
+
+    public void doCreateInstance(AzureCreateVirtualNetworkRef azureCreateVirtualNetworkRef, AzureUser azureUser)
+            throws FogbowException {
+
+        Azure azure = AzureClientCacheManager.getAzure(azureUser);
+        Observable<Indexable> virtualNetworkCreationObservable = buildVirtualNetworkCreationObservable(azureCreateVirtualNetworkRef, azure);
+        subscribeVirtualNetworkCreation(virtualNetworkCreationObservable);
+    }
+
+    /**
+     * Build full virtual network creation process behaviour that it consists in:
+     * 1 - Create Security Group
+     * 2 - Create Virtual Network based on security group created previously
+     */
+    @VisibleForTesting
+    Observable<Indexable> buildVirtualNetworkCreationObservable(AzureCreateVirtualNetworkRef azureCreateVirtualNetworkRef, Azure azure) {
+        Observable<Indexable> securityGroupObservable = buildCreateSecurityGroupObservable(azureCreateVirtualNetworkRef, azure);
+        return securityGroupObservable
+                .doOnNext(indexableSecurityGroup -> {
+                    LOGGER.info(Messages.Info.FIRST_STEP_CREATE_VNET_ASYNC_BEHAVIOUR);
+                    doNetworkCreationStepTwoSync(indexableSecurityGroup, azureCreateVirtualNetworkRef, azure);
+                    LOGGER.info(Messages.Info.SECOND_STEP_CREATE_VNET_ASYNC_BEHAVIOUR);
+                })
+                .onErrorReturn(error -> {
+                    LOGGER.error(Messages.Error.ERROR_CREATE_VNET_ASYNC_BEHAVIOUR);
+                    return null;
+                })
+                .doOnCompleted(() -> {
+                    LOGGER.info(Messages.Info.END_CREATE_VNET_ASYNC_BEHAVIOUR);
+                });
+    }
+
+    @VisibleForTesting
+    Observable<Indexable> buildCreateSecurityGroupObservable(AzureCreateVirtualNetworkRef azureCreateVirtualNetworkRef, Azure azure) {
+        String name = azureCreateVirtualNetworkRef.getName();
+        String resourceGroupName = azureCreateVirtualNetworkRef.getResourceGroupName();
+        String cidr = azureCreateVirtualNetworkRef.getCidr();
+        Region region = Region.fromName(this.regionName);
+
+        return AzureNetworkSDK.createSecurityGroupAsync(azure, name, region, resourceGroupName, cidr);
+    }
+
+
+    @VisibleForTesting
+    void doNetworkCreationStepTwoSync(Indexable indexableSecurityGroup,
+                                           AzureCreateVirtualNetworkRef azureCreateVirtualNetworkRef,
+                                           Azure azure) {
+
+        NetworkSecurityGroup networkSecurityGroup = (NetworkSecurityGroup) indexableSecurityGroup;
+        String name = azureCreateVirtualNetworkRef.getName();
+        String resourceGroupName = azureCreateVirtualNetworkRef.getResourceGroupName();
+        String cidr = azureCreateVirtualNetworkRef.getCidr();
+        Region region = Region.fromName(this.regionName);
```

Comment:
The **fromName** method may behave differently as expected. This method will search for a region name; if the region is not found, a new region with that name will be created. 

I suggest using **Region.findByLabelOrName** method.

Reply:
Done.
Note: I have fixed it in the last branch of Azure Network context.